### DISCUSSION TITLE
next-auth github provider problem

### DISCUSSION THREAD

**Discussion by niklasrammelkamp** [neuefische/questions#519] [Web] [answered]
Hi there,
yesterday I build in a next-auth authentification with GitHub. It's working quite fine. 
I have two questions about it. 

The first one:
When I log in it's working all good, when I log out, everything perfect. But when I wanna log in again And choose GitHub I automatically get logged in with my last account. That's nice I know but maybe I wanna create a new account and that seems to be impossible if I wanna stay in the same browser or device. 
So how can I create a new account if I already have been logged in?

The second question is:
When I send my preview link to someone else and he or she wants to log in, the GitHub provider always wants to log in as me. I have a Screenshot of what it looks like.
How can I change this?

![A8DFC70C-7D29-4139-BB78-BAC2615ED3E9](https://user-images.githubusercontent.com/118187307/217771163-12259393-2745-4bfb-8d10-7f7d83d78584.png)

Thanks for helping me out.

**Comment by RobinWitt** [comment]
Hi Niklas, as for the second question: 
I think it is actually correct, just a little bit confusing. You are the owner/admin of the app, therefore everyone that wants to log in needs to "authorize" you. In a company, it would be more likely that the GitHub account has the name of the company or organization, so the button name would be less confusing in that case. From what I found it is not possible to change that, other than giving the ownership of your app to another account or changing your GitHub account name.
Hope this helps!

**Reply by RobinWitt** [reply]
Your first question is a bit complex, since you have to change the default `signIn` and session handling process. You would likely need to create a custom redirect page where the user does not only have the option to click on the "provider" but also have a button to "switch account" (deletes session cookies, see below):

[see repo here](https://github.com/nextauthjs/next-auth/blob/main/packages/next-auth/src/react/index.tsx)
line 279
[doc here](https://next-auth.js.org/getting-started/client#nextauthjs--react-query)

```jsx
/**
 * Signs the user out, by removing the session cookie.
 * Automatically adds the CSRF token to the request.
 *
 * [Documentation](https://next-auth.js.org/getting-started/client#signout)
 */
export async function signOut<R extends boolean = true>(
  options?: SignOutParams<R>
): Promise<R extends true ? undefined : SignOutResponse> {
  const { callbackUrl = window.location.href } = options ?? {}
  const baseUrl = apiBaseUrl(__NEXTAUTH)
  const fetchOptions = {
    method: "post",
    headers: {
      "Content-Type": "application/x-www-form-urlencoded",
      "X-Auth-Return-Redirect": "1",
    },
    body: new URLSearchParams({
      csrfToken: (await getCsrfToken()) ?? "",
      callbackUrl,
    }),
  }
  const res = await fetch(`${baseUrl}/signout`, fetchOptions)
  const data = await res.json()

  broadcast.post({ event: "session", data: { trigger: "signout" } })

  if (options?.redirect ?? true) {
    const url = data.url ?? callbackUrl
    window.location.href = url
    // If url contains a hash, the browser does not reload the page. We reload manually
    if (url.includes("#")) window.location.reload()
    // @ts-expect-error
    return
  }

  await __NEXTAUTH._getSession({ event: "storage" })

  return data
}


```

**Reply by RobinWitt** [reply]
Of course, the code is only an example, you would have to change it for your use case and framework.

**Comment by doemser** [comment, marked as answer]
## The first one:
### You can try two things:
1. In the browser you want to use, log out from the github account you dont want to be automatically taken for login
 or
2. Unauthorize the app
<img width="1284" alt="Screenshot 2023-02-09 at 15 00 42" src="https://user-images.githubusercontent.com/93415777/217833734-8548259a-f1c3-4d39-9fe3-0fcf162b5409.png">

## The second one:
foto_app_preview by niklasrammelkamp wants to access Robin Witts data. Looks correct to me.

**Reply by doemser** [reply]
Whoops I circled the wrong tab, you should look for the app under "Authorized OAuth Apps"
